pkg/system/resources.go

```diff
@@ -1,76 +1,73 @@
 
-	buildkitdMemoryRequest = *resource.NewQuantity(256*mi, resource.BinarySI)    // BUILDKITD_MEMORY_REQUEST
-	buildkitdMemoryLimit   = *resource.NewQuantity(10*gi, resource.BinarySI)     // BUILDKITD_MEMORY_LIMIT
-	buildkitdCPURequest    = *resource.NewMilliQuantity(800, resource.DecimalSI) // BUILDKITD_CPU_REQUEST
+// ValidateResources is used by the CLI to validate that the values passed
+// can be parsed as a resource.Quantity.
+func ValidateResources(resources ...string) error {
+	for _, rs := range resources {
+		if rs == "" {
+			continue
+		}
 
-	buildkitdServiceMemoryRequest = *resource.NewQuantity(128*mi, resource.BinarySI)    // BUILDKITD_SERVICE_MEMORY_REQUEST
-	buildkitdServiceMemoryLimit   = *resource.NewQuantity(256*mi, resource.BinarySI)    // BUILDKITD_SERVICE_MEMORY_LIMIT
-	buildkitdServiceCPURequest    = *resource.NewMilliQuantity(200, resource.DecimalSI) // BUILDKITD_SERVICE_CPU_REQUEST
-)
+		req, limit, both := strings.Cut(rs, ":")
+		if !both {
+			_, err := resource.ParseQuantity(rs)
+			return err
+		}
 
-func RegistryResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_REQUEST", registryMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("REGISTRY_CPU_REQUEST", registryCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_LIMIT", registryMemoryLimit),
-		},
+		_, err := resource.ParseQuantity(req)
+		if err != nil {
+			return err
+		}
+		_, err = resource.ParseQuantity(limit)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func BuildkitdResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_MEMORY_REQUEST", buildkitdMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("BUILDKITD_CPU_REQUEST", buildkitdCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_MEMORY_LIMIT", buildkitdMemoryLimit),
-		},
-	}
-}
+// ResourceRequirementsFor is used by components to create a ResourceRequirements struct
+// based on strings found in the apiv1.Config struct.
+func ResourceRequirementsFor(memory, cpu string) corev1.ResourceRequirements {
+	requirements := corev1.ResourceRequirements{Requests: corev1.ResourceList{}, Limits: corev1.ResourceList{}}
 
-func BuildkitdServiceResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_SERVICE_MEMORY_REQUEST", buildkitdServiceMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("BUILDKITD_SERVICE_CPU_REQUEST", buildkitdServiceCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_SERVICE_MEMORY_LIMIT", buildkitdServiceMemoryLimit),
-		},
+	var resources = map[corev1.ResourceName]string{
```

Comment:
nit:
```suggestion
	resources := map[corev1.ResourceName]string{
```

Reply:
Slight deviation but I believe I captured the essence of what you were looking for here!